src/urllib3/contrib/securetransport.py

```diff
@@ -396,11 +402,43 @@ def _custom_validate(self, verify, trust_bundle):
         Called when we have set custom validation. We do this in two cases:
         first, when cert validation is entirely disabled; and second, when
         using a custom trust DB.
+        Raises an SSLError if the connection is not trusted.
         """
         # If we disabled cert validation, just say: cool.
         if not verify:
             return
 
+        successes = (
+            SecurityConst.kSecTrustResultUnspecified,
+            SecurityConst.kSecTrustResultProceed,
+        )
+        try:
+            trust_result = self._evaluate_trust(trust_bundle)
+            if trust_result in successes:
+                return
+            reason = "error code: %d" % (trust_result,)
+        except Exception as e:
+            # Do not trust on error
+            reason = "exception: %r" % (e,)
+
+        # SecureTransport does not send an alert nor shuts down the connection.
+        rec = build_tls_alert_record(
+            self.version(), TLS_ALERT_SEVERITY_FATAL, TLS_ALERT_DESCRIPTION_UNKNOWN_CA
+        )
+        self.socket.sendall(rec)
+        # close the connection immediately
+        opts = struct.pack("ii", 1, 0)
+        self.socket.setsockopt(socket.SOL_SOCKET, socket.SO_LINGER, opts)
```

Comment:
Can we have a comment about the struct that `SO_LINGER` accepts so the `struct.pack()` call makes more sense?

Reply:
Added 👍